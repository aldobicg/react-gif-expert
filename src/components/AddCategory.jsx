import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');
  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onFormSubmit = (event) => {
    event.preventDefault();
    const inputValueTrimmed = inputValue.trim();
    if (inputValueTrimmed.length <= 1) return;
    onNewCategory(inputValueTrimmed);
    setInputValue('');
  };

  return (
    <form onSubmit={onFormSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
