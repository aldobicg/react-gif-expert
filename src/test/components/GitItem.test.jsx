import React from 'react';
import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('GifItem', () => {
  const title = 'Test Title';
  const url = 'https://test.url';
  it('should render correctly with given props', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toContain(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole('img');
    expect(src).toContain(url);
    expect(alt).toContain(alt);
  });
  it('debe de mostart el titulo', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
