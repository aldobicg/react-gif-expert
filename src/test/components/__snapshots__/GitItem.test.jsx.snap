// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should render correctly with given props 1`] = `
<div
  class="card"
>
  <img
    alt="Test Title"
    src="https://test.url"
  />
  <p>
    Test Title
  </p>
</div>
`;
